### PR TITLE
querier: Moved query range API to middlewares reusing Cortex cache.

### PR DESCRIPTION
**Pull request by bwplotka**
Very WIP

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,26 @@
 module github.com/improbable-eng/thanos
 
+replace github.com/cortexproject/cortex => github.com/bwplotka/cortex v0.0.0-20190416102825-ae4098d14a9c
```
Comment (bwplotka) [inline go.mod]: Tmp as per: https://github.com/cortexproject/cortex/pull/1332

File: pkg/query/api.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+type API struct {
+	engine *promql.Engine
+
+	replicaLabel string
+	storeAPI     storepb.StoreServer
+}
+
+func NewAPI(engine *promql.Engine, replicaLabel string, storeAPI storepb.StoreServer) *API {
```
Comment (bwplotka) [inline pkg/query/api.go]: I think I will factor this out to just use PromQL natively in `api.go` and new middlewares.

**Comment by GiedriusS** [comment]
@bwplotka why did you close? Do you intend to continue working on this in a separate branch/PR? This could be a nice starting point for someone working on this.

**Comment by bwplotka** [comment]
Sorry for not commenting when closing. I will respond on the issue: https://github.com/thanos-io/thanos/issues/1006

**Pull request closed without merging**